### ISSUE TITLE
Improve specification of r_gov_shift and r_gov_scale: documentation and bounds 

### ISSUE THREAD

**Issue by rickecon** [PSLmodels/OG-Core#841]
[Equation (79)](https://pslmodels.github.io/OG-Core/content/theory/government.html#equation-equnbalgbc-rate-wedge) in the OG-Core documentation specifies the exogenous wedge between the government interest rate `r_{gov,t}` and the private return or marginal product of capital `r_t`.

r_{gov,t} = (1 - tau_{d,t}) * r_t - mu_d

We have two parameters in the model, `r_gov_shift` and `r_gov_scale` to represent the two parameters `tau_{d,t}` and `mu_d`. These are specified and defined correctly in [`constants.py` lines 229-236](https://github.com/PSLmodels/OG-Core/blob/master/ogcore/constants.py#L229). Also, references and usage in `test_basic.py` lines [71](https://github.com/PSLmodels/OG-Core/blob/master/tests/test_basic.py#L71) and [128](https://github.com/PSLmodels/OG-Core/blob/master/tests/test_basic.py#L128) and in [`test_fiscal.py` lines 247-254](https://github.com/PSLmodels/OG-Core/blob/master/tests/test_fiscal.py#L247) are fine.

@SeaCelo and I found the following three issues with how these two parameters are specified in the documentation and used in the model. Each of these needs the recommended update. Problems (1) and (2) are the only ones that apply directly to OG-Core. But problem (3) requires updates to country calibrations like OG-USA.

(1) The specification in [`fiscal.get_r_gov()`](https://github.com/PSLmodels/OG-Core/blob/master/ogcore/fiscal.py#L353) is the following:
```python
r_gov = np.maximum(p.r_gov_scale * r - p.r_gov_shift, 0.00)
```
So we should change the specification in equation (79) of the documentation to be consistent with the model `get_r_gov()`.

r_{gov,t} = max( tau_{d,t} * r_t - mu_d, 0)

(2) Because the model is specified as in [`fiscal.get_r_gov()`](https://github.com/PSLmodels/OG-Core/blob/master/ogcore/fiscal.py#L353), the bounds specified in [`default_parameters.json`](https://github.com/PSLmodels/OG-Core/blob/master/ogcore/default_parameters.json) should be the following,
```
r_gov_scale = [0.0, 1.0]
r_gov_shift = [0.0, 0.3]
```
but they are actually the following:
```
r_gov_scale = [0.0, 4.0]
r_gov_shift = [0.0, 0.3]
```
It seems that the bounds in `default_parameters.json` were set based on the current equation (79) in the documentation as if `r_gov_scale = mu_d` and `r_gov_shift = tau_{d,t}`. Things just worked out by coincidence because the default parameter values were set to be within the incorrectly specified bounds for `r_gov_scale`.

We should change the bounds of `r_gov_scale` to `r_gov_scale = [0.0, 1.0]` with the default value staying the same at 1.0.

(3) In OG-USA calibration module [`macro_params.py` lines 104-117](https://github.com/PSLmodels/OG-USA/blob/master/ogusa/macro_params.py#L104), `r_gov_scale` and `r_gov_shift` are calibrated by a regression that switches their values to represent the opposite parameters in [equation (79)](https://pslmodels.github.io/OG-Core/content/theory/government.html#equation-equnbalgbc-rate-wedge) of the OG-Core documentation such that `r_gov_scale=mu_d` and `r_gov_shift=tau_{d,t}`. This is because the `macro_params.py` calibration in OG-USA runs a regression of the 10-year Treasury rate (`r_{gov,t}`) on a constant and on the BAA Corporate Bond Rate (`r_t`). `r_gov_shift` is defined as the regression coefficient on the private interest rate and `r_gov_scale` is defined as the estimated constant. [Lines 116 and 117](https://github.com/PSLmodels/OG-USA/blob/master/ogusa/macro_params.py#L116) in `macro_params.py` in OG-USA should be switched to be the following including putting a negative sign on the constant regression coefficient (as well as in any other country calibration).
```python
macro_parameters["r_gov_scale"] = np.minimum(np.maximum(res.params["BAA Corp Bond Rates"], 0.0), 1.0)
macro_parameters["r_gov_shift"] = np.minimum(np.maximum(-res.params["constant"], 0.0), 0.3)
```
Note that our specification of the relationship between `r_{gov,t}` and `r_t` only allows the following two families of relationships for figure 1 where `mu_d=0` and figure 2 where `mu_d in (0, 0.3]`
![IMG_3634](https://user-images.githubusercontent.com/4317690/210683792-f88e7756-2da3-4221-9408-fc6f162040e6.jpg)

cc: @jdebacker @SeaCelo

**Comment by rickecon** [comment]
@jdebacker.

Re (1): I think my preference would be to update the documentation to reflect the zero lower bound. But I am OK with leaving it as it is. But I definitely think we should change the formulation in the documentation to be `tau * r` rather than `(1-tau) * r`.

Re (2): I think you are right that the restrictions I was assuming were too strong. Here are some guiding principles for some restrictions on that relationship.
* I know you don't propose this, but let me start with this point. We could fit a nice higher order polynomial or nonparametric function `r_{g,t} = f(r_t)` to match the interest rate data well over the range of private and government interest rate data. But that will make the government interest rate fluctuate for reasons that are impossible to explain. And that government interest rate controls a lot of the variables we care about, like government debt and its effect on the other macro variables through crowding out. We definitely want something simpler that captures the "average" relationship between `r_gov` and `r_t`.
* So I think linear relationship with a zero lower bound on `r_gov` is a good idea because it is a good characterization of an average relationship. But I do think we should have further limits on what `tau` and `mu` can be in that linear relationship, even if we impose a zero lower bound. Here are some restrictions that make sense to me:
    a) `tau>=0` always. It doesn't make sense to me to ever have a negative slope `tau<0`
    b) If we estimate a negative `mu<0` (positive y-intercept), we should impose that `tau<=1`
    c) `tau` can only be greater than one `tau>1` if `mu` is positive `mu>0`(negative y-intercept, excluding zero lower bound).
    d) It seems intuitive that we should have upper and lower bounds on `tau`, but I don't know what they should be. Maybe `tau <= 2.0` and `tau >=-0.5`. But I just pulled those out of the intuition hat.
    e) It seems like we should have some limits on `mu`, but those are even harder to pin down.
    f) A solution to my issue in (e) could be to run a normalized regression (transformed into percent deviations, mean zero, variance 1) so that the estimated equation captures the central relationship of the data. This would help if the interest rate units of the model live in a slightly different place than the interest rate units of the data. For example, the mean `r_t` of the model might be higher than the mean `r_t` in the data. We want the relationship of `r_gov` around the mean `r_t` in the model to mimic the relationship that exists around the mean of the data.

**Comment by SeaCelo** [comment]
@rickecon @jdebacker: In case it is helpful to narrow down the range of both the `shift` and `scale` estimates, there is a recent paper that estimated the relationship for both advanced and emerging economies (see the regression results as well as summary statistics on table 7, table 8. Figure 5 is their model of the relationship). [I mentioned it and another paper here.](https://github.com/EAPD-DRB/OG-ZAF/issues/22#issue-1492291828)

It is notable that both the intercept and the coefficient are always positive. This tracks with what I remember is a theoretical expectation (sovereign risk is always lower, on average, than private risk). It also fits the empirical findings as these two papers show.

As for the specific range for the variables, the `shift` parameter should have a wider range if this model is to be applied to lower income countries (as we are doing for ZAF and in the future other developing countries. The credit markets in this countries can be expected to show much higher spreads than in emerging and advanced markets.

At the very least, perhaps OG-CORE can allow the country calibration to decide based on data rather than hard-coding this limit.

**Comment by SeaCelo** [comment]
@rickecon I may be misreading this, but I don't think inverting the coefficient sign is necessary unless the fitted regression is giving the wrong sign. In the specification in [fiscal.get_r_gov()](https://github.com/PSLmodels/OG-Core/blob/master/ogcore/fiscal.py#L353), the coefficient term is already subtracted. 
`r_gov = np.maximum(p.r_gov_scale * r - p.r_gov_shift, 0.00)`

I also think the range for `r_gov_shift` should be much higher than the proposed 0.3 value, and assuming that the switching of the parameters is corrected. It would then be:
```
r_gov_scale = [0.0, 1.0]  #tau
r_gov_shift = [0.0, 4.0]  #mu, but could be even higher
```
So:
`macro_parameters["r_gov_shift"] = np.minimum(np.maximum(res.params["constant"], 0.0), 4.0)`

**Comment by rickecon** [comment]
This issue is temporarily resolved by PR #844. This PR changed the minval of `r_gov_shift` to -0.3. We still might consider requiring that `r_gov <= r_t` always by implementing a `min` operator in `fiscal.get_r_gov()`. But we are pushing that down the road. We don't need it right now.
